### PR TITLE
Support custom oauth bearer

### PR DESCRIPTION
**Pull request by mensfeld**
waterdrop part of https://github.com/karafka/karafka/issues/1767

additionally I added extra specs that were missing and missing config rules.

### CODE REVIEW

File: lib/waterdrop/clients/rdkafka.rb [hunk 1]
```diff
@@ -11,20 +11,41 @@ class << self
...
         # @note We overwrite this that way, because we do not care
         def new(producer)
-          config = producer.config.kafka.to_h
+          kafka_config = producer.config.kafka.to_h
+          monitor = producer.config.monitor
 
-          client = ::Rdkafka::Config.new(config).producer
+          client = ::Rdkafka::Config.new(kafka_config).producer(native_kafka_auto_start: false)
+
+          # Register statistics runner for this particular type of callbacks
```
Comment (nijikon) [inline lib/waterdrop/clients/rdkafka.rb:19]: oauthbearer not statistics

**Review by nijikon — APPROVED** [review]
LGTM

**Comment by bruce-szalwinski-he** [comment]
works great!

```
      producer = WaterDrop::Producer.new do |config|
        config.deliver = true
        config.kafka = kafka_config
      end

      producer.monitor.subscribe('oauthbearer.token_refresh') do |event|
        OAuthTokenRefresher.new.on_oauthbearer_token_refresh(event)
      end
```

```
class OAuthTokenRefresher
  def on_oauthbearer_token_refresh(event)
    signer = AwsMskIamSaslSigner::MSKTokenProvider.new(region: 'us-east-1')
    token = signer.generate_auth_token

    event[:bearer].oauthbearer_set_token(
      token: token.token,
      lifetime_ms: token.expiration_time_ms,
      principal_name: 'kafka-cluster'
    )
  end
end
```

**Review by bruce-szalwinski-he — APPROVED** [review]

**Comment by mensfeld** [comment]
@bruce-szalwinski-he I reworked the issues you had. Now it should be good.

**Pull request merged**